src/System.Text.Json/tests/JsonElementValueTests.cs

```diff
@@ -0,0 +1,90 @@
+
+
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+namespace System.Text.Json.Tests
+{
+    public static class JsonElementValueTests
+    {
+        [Fact]
+        public static void ValueEquals_EmptyJsonString_True()
+        {
+            const string jsonString = "\"\"";
+            JsonElement jElement = (JsonElement)JsonDocument.Parse(jsonString).RootElement.Clone();
+
+            Assert.True(jElement.ValueEquals(""));
+            Assert.True(jElement.ValueEquals("".AsSpan()));
+            Assert.True(jElement.ValueEquals(default(ReadOnlySpan<char>)));
+            Assert.True(jElement.ValueEquals(default(ReadOnlySpan<byte>)));
+        }
+
+        [Fact]
+        public static void ValueEquals_StackallocThreshold_True()
+        {
+            string expected = new string('ѧ', 257);
+            var jsonString = $"\"{expected}\"";
+            JsonElement jElement = (JsonElement)JsonDocument.Parse(jsonString).RootElement.Clone();
+
+            Assert.True(jElement.ValueEquals(expected));
+            Assert.True(jElement.ValueEquals(expected.AsSpan()));
+            byte[] expectedGetBytes = Encoding.UTF8.GetBytes(expected);
+            Assert.True(jElement.ValueEquals(expectedGetBytes));
+        }
+
+        [Theory]
+        [InlineData("\"conne\\u0063tionId\"", "connectionId")]
+        [InlineData("\"connectionId\"", "connectionId")]
+        [InlineData("\"123\"", "123")]
+        [InlineData("\"My name is \\\"Ahson\\\"\"", "My name is \"Ahson\"")]
+        public static void ValueEquals_JsonTokenStringType_True(string jsonString, string expected)
+        {
+            JsonElement jElement = (JsonElement)JsonDocument.Parse(jsonString).RootElement.Clone();
+            Assert.True(jElement.ValueEquals(expected));
+            Assert.True(jElement.ValueEquals(expected.AsSpan()));
+            byte[] expectedGetBytes = Encoding.UTF8.GetBytes(expected);
+            Assert.True(jElement.ValueEquals(expectedGetBytes));
+        }
+
+        [Theory]
+        [InlineData("\"hello\"", new char[1] { (char)0xDC01 })]    // low surrogate - invalid
+        [InlineData("\"hello\"", new char[1] { (char)0xD801 })]    // high surrogate - missing pair
+        public static void InvalidUTF16Search(string jsonString, char[] lookup)
+        {
+            JsonElement jElement = (JsonElement)JsonDocument.Parse(jsonString).RootElement.Clone();
+            Assert.False(jElement.ValueEquals(lookup));
+        }
+
+        [Theory]
+        [InlineData("\"connectionId\"", "\"conne\\u0063tionId\"")]
+        [InlineData("\"conne\\u0063tionId\"", "bonnectionId")]
```

Comment:
nit: Maybe add a comment highlighting the string is different intentionally. At a quick glance, I thought the `b` in `bonnectionId` was a typo.